--- a/upstream/libtailscale/swift/TailscaleKit/Listener.swift
+++ b/upstream/libtailscale/swift/TailscaleKit/Listener.swift
@@ -53,7 +53,7 @@ public actor Listener {

     deinit {
         if listener != 0 {
-            unistd.close(listener)
+            Darwin.close(listener)
         }
     }

@@ -61,7 +61,7 @@ public actor Listener {
     /// Listeners will be closed automatically on deallocation
     public func close() {
         if listener != 0 {
-            unistd.close(listener)
+            Darwin.close(listener)
             listener = 0
         }
         _state = .closed
