--- a/upstream/libtailscale/swift/TailscaleKit/OutgoingConnection.swift
+++ b/upstream/libtailscale/swift/TailscaleKit/OutgoingConnection.swift
@@ -79,7 +79,7 @@ public actor OutgoingConnection {

     deinit {
         if conn != 0 {
-            unistd.close(conn)
+            Darwin.close(conn)
         }
     }

@@ -88,7 +88,7 @@ public actor OutgoingConnection {
     /// state to .closed
     public func close() {
         if conn != 0 {
-            unistd.close(conn)
+            Darwin.close(conn)
             conn = 0
         }
         state = .closed
@@ -102,7 +102,7 @@ public actor OutgoingConnection {
             throw TailscaleError.connectionClosed
         }

-        let bytesWritten = unistd.write(conn, data.withUnsafeBytes { $0.baseAddress! }, data.count)
+        let bytesWritten = Darwin.write(conn, data.withUnsafeBytes { $0.baseAddress! }, data.count)

         if bytesWritten != data.count {
             throw TailscaleError.shortWrite
