--- a/upstream/libtailscale/tailscale.c
+++ b/upstream/libtailscale/tailscale.c
@@ -23,6 +23,7 @@ extern int TsnetGetIps(int sd, char *buf, size_t buflen);
 extern int TsnetGetRemoteAddr(int listener, int conn, char *buf, size_t buflen);
 extern int TsnetListen(int sd, char* net, char* addr, int* listenerOut);
 extern int TsnetAccept(int ld, int* connOut);
+extern int TsnetListenPacket(int sd, char* net, char* addr, int* listenerOut);
 extern int TsnetLoopback(int sd, char* addrOut, size_t addrLen, char* proxyOut, char* localOut);
 extern int TsnetEnableFunnelToLocalhostPlaintextHttp1(int sd, int localhostPort);

@@ -54,6 +55,10 @@ int tailscale_accept(tailscale_listener ld, tailscale_conn* conn_out) {
 	return TsnetAccept(ld, (int*)conn_out);
 }

+int tailscale_listen_packet(tailscale sd, const char* network, const char* addr, tailscale_listener* listener_out) {
+	return TsnetListenPacket(sd, (char*)network, (char*)addr, (int*)listener_out);
+}
+
 int tailscale_getremoteaddr(tailscale_listener l, tailscale_conn conn, char* buf, size_t buflen) {
 	return TsnetGetRemoteAddr(l, conn, buf, buflen);
 }
